ourhexenv.py

```diff
@@ -57,14 +57,25 @@ def __init__(self, board_size=5, sparse_flag=True):
 
     def reset(self):
         self.board = np.zeros((self.board_size, self.board_size), dtype=int)
+        
+        self.isFirst = True
         self.is_pie_rule_usable = False
         self.agent_selection = "player_1"
+
         self.dones = {agent: False for agent in self.agents}
         self.infos = {agent: [] for agent in self.agents}
-
         self._cumulative_rewards = {agent: 0 for agent in self.agents}
         self.terminations = {agent: False for agent in self.agents}
         self.truncations = {agent: False for agent in self.agents}
+        self.rewards = {agent: 0 for agent in self.agents}
+
+        self.action_space = spaces.Discrete(self.board_size * self.board_size + 1)
+        self.action_spaces = {agent: self.action_space for agent in self.agents}
```

Comment:
There is no need to reset the action spaces as they remain constant and unchanged during the lifetime of the environment

Reply:
Got it thanks. I will modify that.